src/components/RepoBadge/RepoBadge.js

```diff
@@ -0,0 +1,73 @@
+import React, { useMemo } from 'react'
+import { blockExplorerUrl, Link } from '@aragon/ui'
+import { network } from '../../environment'
+import AppIcon from '../../components/AppIcon/AppIcon'
+import { KNOWN_ICONS, isKnownRepo } from '../../repo-utils'
+import { repoBaseUrl, parseHub } from '../../url-utils'
+
+function useRepoBadge(repo) {
+  const {
+    content: { appId, contractAddresses, name, icons },
+    version,
+  } = repo
+
+  return {
+    baseUrl: repoBaseUrl(repo),
+    icons: icons,
+    contractAddress: contractAddresses,
+    knownIcon: KNOWN_ICONS.has(appId) ? KNOWN_ICONS.get(appId) : null,
+    name: name,
+  }
+}
+
+const RepoBadge = ({ repo }) => {
+  const { baseUrl, icons, contractAddress, knownIcon, name } = useRepoBadge(
+    repo
+  )
+
+  return (
+    <div
+      css={`
+        display: inline-flex;
+        align-items: center;
+        min-width: 0;
+      `}
+    >
+      <Link
+        target="_blank"
+        href={blockExplorerUrl('address', contractAddress, {
```

Comment:
`contractAddress` seems to always be null (see above with `repoBaseUrl()`)

Reply:
Yep, but it has to do more with `contractAddresses != contractAddress`, there was a typo in there.